docs/superpowers/plans/2026-06-08-dungeon-graph-2a.md

```
@@ -0,0 +1,53 @@
+
+**Goal:** Stand up the data-driven dungeon graph engine — `levels.toml`, a `Dungeon` (graph + generated-level cache), portal-stair travel with persistence, and per-level spawn tables — proven with 3 real levels. See `docs/superpowers/specs/2026-06-08-tsl-dungeon-graph-design.md`.
+
+**Architecture:** `content.LevelDef` (loaded from `levels.toml`) describes each level; `gen.LevelFromDef` builds one; `game.Dungeon` holds the graph + cache + current id and is injected with the `gen` builder by `cmd`; `Game.Travel` moves between levels via `Portal` tiles, persisting visited levels. Replaces `Depth`/`Descend`/`NewLevelFn`/`gen.Rooms(...,depth)`.
+
+**Tech Stack:** Go 1.21 (`GOTOOLCHAIN=local`). Commands run from the `go/` directory at the repo root, prefixed `GOTOOLCHAIN=local`; prepend `env -u GOROOT` on a GOROOT error. TDD, commit per task, suite green between commits.
+
+---
+
+## Task 1: `content.LevelDef` + `levels.toml` (content)
+**Files:** `internal/content/content.go`, `internal/content/content_test.go`
+- Types: `SpawnEntry{Monster string `toml:"monster"`; Weight int `toml:"weight"`}`, `LevelDef{ID string `toml:"-"`; Name string; W int `toml:"width"`; H int `toml:"height"`; Start bool `toml:"start"`; Links []string `toml:"links"`; Monsters int `toml:"monsters"`; Spawn []SpawnEntry `toml:"spawn"`; Win bool `toml:"win"`}`. `Content.Levels map[string]*LevelDef`. `levelsFile{Level map[string]*LevelDef `toml:"level"`}`.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)